### DISCUSSION TITLE
How does Helix handle sequences?

### DISCUSSION THREAD

**Discussion by test3211234** [helix-editor/helix#8815] [Q&A] [answered]
I've looked through so much of Helix's code trying to find how they handle key sequences with a timeout. For example, press a and then press b in succession within 500ms. Or is it that Helix doesn't have these, and all the key sequences don't have timeouts? If you were to have these, how would you go about implementing them? Thanks!

**Comment by pascalkuthe** [comment, marked as answer]
There are no timeouts. Keymap related code is in helix-term (::ui::editor and ::keymap)

**Reply by test3211234** [reply]
Ok, thanks. If there were timeouts, how would you implement them?
